File: Reading/数据结构与算法图解/code/solutions/solution12-1.py
def addUntil100(array):
  if len(array) == 0:
    return 0

  sumOfRemainingNumbers = addUntil100(array[1:])

  if array[0] + sumOfRemainingNumbers > 100:
    return sumOfRemainingNumbers
  else:
    return array[0] + sumOfRemainingNumbers

File: Reading/数据结构与算法图解/code/solutions/solution12-1.rb
#---
# Excerpted from "A Common-Sense Guide to Data Structures and Algorithms, Second Edition",
# published by The Pragmatic Bookshelf.
# Copyrights apply to this code. It may not be used to create training material,
# courses, books, articles, and the like. Contact us if you are in doubt.
# We make no guarantees that this code is fit for any purpose.
# Visit http://www.pragmaticprogrammer.com/titles/jwdsal2 for more book information.
#---
def add_until_100(array)
  return 0 if array.length == 0
  sum_of_remaining_numbers = add_until_100(array[1, array.length - 1])
  if array[0] + sum_of_remaining_numbers > 100
    return sum_of_remaining_numbers
  else
    return array[0] + sum_of_remaining_numbers
  end
end